utils/minnesota.py

```diff
@@ -0,0 +1,423 @@
+                "recipient_type",
+                "state",
+                "party",
+                "company",
+            ],
+        )
+        ind_recipient_df = ind_recipient_df.rename(
+            columns={
+                "recipient_id": "id",
+                "recipient_first_name": "first_name",
+                "recipient_last_name": "last_name",
+                "recipient_full_name": "full_name",
+                "recipient_type": "entity_type",
+            }
+        )
+        ind_donor_df = pd.DataFrame(
+            data=df[df["donor_type"].isin(["Individual", "Lobbyist"])].drop_duplicates(
+                subset="donor_id"
+            ),
+            columns=[
+                "donor_id",
+                "donor_first_name",
+                "donor_last_name",
+                "donor_full_name",
+                "donor_type",
+                "state",
+                "party",
+                "company",
+            ],
+        )
+        ind_donor_df = ind_donor_df.rename(
+            columns={
+                "donor_id": "id",
+                "donor_first_name": "first_name",
+                "donor_last_name": "last_name",
+                "donor_full_name": "full_name",
+                "donor_type": "entity_type",
+            }
+        )
+        ind_df = pd.concat([ind_recipient_df, ind_donor_df], ignore_index=True)
+
+        # Create organization table from both recipient and donor entries
+        org_recipient_df = pd.DataFrame(
+            data=df[
+                ~df["recipient_type"].isin(["Individual", "Lobbyist"])
+            ].drop_duplicates(subset="recipient_id"),
+            columns=["recipient_id", "recipient_full_name", "state", "recipient_type"],
+        )
+        org_recipient_df = org_recipient_df.rename(
+            columns={
+                "recipient_id": "id",
+                "recipient_full_name": "name",
+                "recipient_type": "entity_type",
+            }
+        )
+        org_donor_df = pd.DataFrame(
+            data=df[~df["donor_type"].isin(["Individual", "Lobbyist"])].drop_duplicates(
+                subset="donor_id"
+            ),
+            columns=["donor_id", "donor_full_name", "state", "donor_type"],
+        )
+        org_donor_df = org_donor_df.rename(
+            columns={
+                "donor_id": "id",
+                "donor_full_name": "name",
+                "donor_type": "entity_type",
+            }
+        )
+        org_df = pd.concat([org_recipient_df, org_donor_df], ignore_index=True)
+
+        tran_df = pd.DataFrame(
+            data=df.drop_duplicates(subset="transaction_id"),
+            columns=[
+                "transaction_id",
+                "donor_id",
+                "year",
+                "amount",
+                "recipient_id",
+                "office_sought",
+                "purpose",
+                "transaction_type",
+            ],
+        )
+
+        return ind_df, org_df, tran_df
```

Comment:
reminder there should be 4 transaction tables as listed in the schema org-org, ind->ind, ind->org, org->ind